helix-term/src/commands.rs

```diff
@@ -1501,6 +1505,57 @@ fn switch_to_lowercase(cx: &mut Context) {
     });
 }
 
+pub fn scroll_page_and_cursor(cx: &mut Context, offset: usize, direction: Direction) {
+    let config = cx.editor.config();
+    let cursor_pos = cx.editor.cursor().0.unwrap();
```

Comment:
this has quite a lot of overhead (its never cached here) I don't think we should we special case the scrolloff this way. vim doesn't do that either

Reply:
As in disregard the scrolloff- just like `move_visual_line_down` does?
This will mean that when we do a C-d from the line 1, we wont do an exact half page scroll in terms of the view. But neovim will actually do an exact half page scrolls in terms of the view

Reply:
hmm I see what you mean, then you need to write your own version of `move_vertically_visual`. That function already calculates the vertical position of the cursor so doing it twice is very inefficient

Reply:
yeh I agree making the scroll distance consistent is probably better- [5a48de0](https://github.com/helix-editor/helix/pull/8015/commits/5a48de0800b4433cf1b8d9bb41bf74d1b5d3a11c)